app/service/planning_svc.py

```diff
@@ -372,18 +372,15 @@ async def _generate_cleanup_links(self, operation, agent, link_status):
         """
         links = []
         for link in [l for l in operation.chain if l.paw == agent.paw]:
-            matched_abilities = await self.get_service('data_svc').locate('abilities',
-                                                                          match=dict(unique=link.ability.unique))
-            if matched_abilities:
-                ability = matched_abilities[0]
-                for cleanup in ability.cleanup:
-                    decoded_cmd = agent.replace(cleanup, file_svc=self.get_service('file_svc'))
-                    variant, _, _ = await self._build_single_test_variant(decoded_cmd, link.used, link.ability.executor)
-                    lnk = Link.load(dict(command=self.encode_string(variant), paw=agent.paw, cleanup=1,
-                                         ability=ability, score=0, jitter=2, status=link_status))
-                    if lnk.command not in [l.command for l in links]:
-                        lnk.apply_id(agent.host)
-                        links.append(lnk)
+            for cleanup in link.executor.cleanup:
+                decoded_cmd = agent.replace(self.encode_string(cleanup), file_svc=self.get_service('file_svc'))
+                variant, _, _ = await self._build_single_test_variant(decoded_cmd, link.used, link.executor.name)
+                lnk = Link.load(dict(command=self.encode_string(variant), paw=agent.paw, cleanup=1,
+                                     ability=link.ability, executor=link.executor, score=0, jitter=2,
+                                     status=link_status))
+                if lnk.command not in [l.command for l in links]:
```

Comment:
I see that this was copied, but it seems like we should be able to perform this check before constructing the `Link` object since it's just thrown away if this condition isn't met. Also membership checks in a list are linear (this applies to other places where we do this) so it'd probably be better to store a `set` of "seen" commands so that the check is faster.